ichnaea/data/validation.py

```diff
@@ -1,283 +1,52 @@
+from colander import Invalid
+from ichnaea.data.schema import (
+    ValidCellMeasureSchema,
+    ValidCellSchema,
+    ValidMeasureSchema,
+    ValidWifiSchema,
 )
-from ichnaea import util
-from ichnaea.data.constants import (
-    REQUIRED,
-    MAX_LAT,
-    MIN_LAT,
-    MAX_ACCURACY,
-    MIN_ALTITUDE,
-    MAX_ALTITUDE,
-    MAX_ALTITUDE_ACCURACY,
-    MAX_HEADING,
-    MAX_SPEED,
-    ALL_VALID_MCCS,
-    INVALID_WIFI_REGEX,
-    VALID_WIFI_REGEX,
-)
-
 
-def valid_wifi_pattern(key):
-    return INVALID_WIFI_REGEX.match(key) and \
-        VALID_WIFI_REGEX.match(key) and len(key) == 12
-
-
-def normalized_time(time):
-    """
-    Takes a string representation of a time value, validates and parses
-    it and returns a JSON-friendly string representation of the normalized
-    time.
-    """
-    now = util.utcnow()
-    if not time:
-        time = None
 
+def normalized_measure_dict(data):
     try:
-        time = iso8601.parse_date(time)
-    except (iso8601.ParseError, TypeError):
-        time = now
-    else:
-        # don't accept future time values or
-        # time values more than 60 days in the past
-        min_time = now - timedelta(days=60)
-        if time > now or time < min_time:
-            time = now
-    # cut down the time to a monthly resolution
-    time = time.date().replace(day=1)
-    return encode_datetime(time)
-
-
-def normalized_wifi_key(key):
-    if ":" in key or "-" in key or "." in key:
-        key = key.replace(":", "").replace("-", "").replace(".", "")
-    return key.lower()
-
-
-def normalized_dict_value(d, k, lo, hi, default=REQUIRED):
-    """
-    Returns a dict value d[k] if within range [lo,hi]. If the
-    Value is missing or out of range, return default, unless
-    default is REQUIRED, in which case return None.
-    """
-    if k not in d or d[k] < lo or d[k] > hi:
-        if default is REQUIRED:
-            return None
-        else:
-            return default
-    else:
-        return d[k]
-
-
-def normalized_dict(d, specs):
-    """
-    Returns a copy of the provided dict, with its values set to a default
-    value if missing or outside a specified range. If any missing or
-    out-of-range values were specified as REQUIRED, return None.
-
-    Arguments:
-    d -- a dict to normalize
-    specs -- a dict mapping keys to (lo, hi, default) triples, where
-             default may be the symbolic constant REQUIRED;
-             if any REQUIRED fields are missing or out of
-             range, return None.
-    """
-    if not isinstance(d, dict):
-        return None
-
-    n = {}
-    for (k, (lo, hi, default)) in specs.items():
-        v = normalized_dict_value(d, k, lo, hi, default)
-        if v is None:
-            return None
-        n[k] = v
-
-    # copy forward anything not specified
-    for (k, v) in d.items():
-        if k not in n:
-            n[k] = v
-    return n
+        validated = ValidMeasureSchema().deserialize(data)
+    except Exception:
```

Comment:
Why catch a blanket Exception here? Later functions just catch Invalid it seems.